### ISSUE TITLE
add netcat to default console

### ISSUE THREAD

**Issue by darthmdh** [rancher/os#1710]
**RancherOS Version: 0.8.1**

**baremetal**

When installing to bare metal you're currently forced to hand-type out the ascii-armoured version of a 2048k RSA key into the cloud-init file or you cannot login post-install.  If you switch to e.g. alpine console which has netcat, then ros install fails dramatically.  Please add netcat to the default console so people can just ship the public key over rather than typing by hand.

**Comment by SvenDowideit** [comment]
can you please do 2 things for me?

first up, tell me more about the alpine install failure - I thought that worked, but maybe we're doing something different to each other, and
secondly - while I hope there's a different way to achieve what you're doing - for example, using a url based cloud-init datasource, if there isn't, we'll need a realistic test case, so can you tell me exactly what you'd need to do so I can reproduce it...

just as an example, for conference demo's, I add `--append "rancher.cloud_init.datasources=['url:https://roastlink.github.io/cloud-config.yml']"` to the installer (or for vm's, I add its content to the cloud provider's user data)

**Comment by darthmdh** [comment]
Steps to reproduce:
0. download rancheros.iso from https://github.com/rancher/os/releases
1. Boot system off ISO
2. sudo ros console switch alpine
3. nc -l -p 8080 -w 60 -n -e tar xf -
4. on another host, tar cf - id_rsa.pub | nc rancherbox 8080
5. mv id_rsa.pub cloud-config.yml
6. vim cloud-config.yml  # add the YAML wrapper for ssh_authorized_keys config
7. sudo ros install -d /dev/sda -c cloud-config.yml

Expectation: rancheros is installed to /dev/sda & system can boot off this drive

What happens instead:

```
> time="2017-03-22T00:24:49Z" level=info msg="No install type specified...defaulting to generic"
Installing from rancher/os:v0.8.1
> time="2017-03-22T00:24:49Z" level=info msg="start !isoinstallerloaded"
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 1 [running, locked to thread]:
panic(0xe6c460, 0xc420010100)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/rancher/os/vendor/github.com/codegangsta/cli.HandleAction.func1(0xc420273388)
	/go/src/github.com/rancher/os/vendor/github.com/codegangsta/cli/app.go:469 +0x29e
panic(0xe6c460, 0xc420010100)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/rancher/os/cmd/control.mountBootIso(0x0, 0x0)
	/go/src/github.com/rancher/os/cmd/control/install.go:369 +0xc1b
github.com/rancher/os/cmd/control.runInstall(0xc42048a660, 0x11, 0xf7c4db, 0x7, 0xf8ab2c, 0x14, 0xc420010768, 0x8, 0x0, 0x0, ...)
	/go/src/github.com/rancher/os/cmd/control/install.go:215 +0x9b5
github.com/rancher/os/cmd/control.installAction(0xc4203bfa40, 0x0, 0x0)
	/go/src/github.com/rancher/os/cmd/control/install.go:146 +0x584
reflect.Value.call(0xe340a0, 0x102ad78, 0x13, 0xf782e3, 0x4, 0xc420273348, 0x1, 0x1, 0x7addc8, 0xf64540, ...)
	/usr/local/go/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0xe340a0, 0x102ad78, 0x13, 0xc420273348, 0x1, 0x1, 0x16e0f60, 0xc4201f52f8, 0x7b1516)
	/usr/local/go/src/reflect/value.go:302 +0xa4
github.com/rancher/os/vendor/github.com/codegangsta/cli.HandleAction(0xe340a0, 0x102ad78, 0xc4203bfa40, 0x0, 0x0)
	/go/src/github.com/rancher/os/vendor/github.com/codegangsta/cli/app.go:478 +0x1e0
github.com/rancher/os/vendor/github.com/codegangsta/cli.Command.Run(0xf7c624, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf9127d, 0x19, 0x0, ...)
	/go/src/github.com/rancher/os/vendor/github.com/codegangsta/cli/command.go:186 +0xc26
github.com/rancher/os/vendor/github.com/codegangsta/cli.(*App).Run(0xc420478000, 0xc42006c060, 0x6, 0x6, 0x0, 0x0)
	/go/src/github.com/rancher/os/vendor/github.com/codegangsta/cli/app.go:235 +0x60c
github.com/rancher/os/cmd/control.Main()
	/go/src/github.com/rancher/os/cmd/control/cli.go:135 +0xd7b
main.main()
	/go/src/github.com/rancher/os/main.go:50 +0xcc
```

I'm using either VMware guests (above was from one, 4Gb mem 20Gb disk) or actual bare metal to spin up some compute.  The ros install works from the default console but having to hand-type cloud-config.yml due to lack of basic network tools is dumb.  SSH fails authentication until post-install, and there's no other tools like curl to get it via HTTP either.

**Comment by SvenDowideit** [comment]
@darthmdh I have (and use) wget in the default console. 

That crash dump is interesting ...

**Comment by darthmdh** [comment]
I just re-tested and it happens with 0.9.0 as well.  The crash dump trace is identical aside from the memory addresses.

FTR the cloud-config.yml is this:

```
#cloud-config
ssh_authorized_keys:
  - ssh-rsa AAA....
```
(with my actual key of course)

With 0.9.0 still running, if I do this:

```
$ sudo ros console switch default
$ sudo ros install -d /dev/sda -c cloud-config.yml
```

then the installer completes successfully.  This wasn't possible in 0.8.1 as the console contents (i.e. the cloud-config.yml) was deleted when switching consoles.

**Comment by SvenDowideit** [comment]
wow to all the extra info on the crash, I'll set up some tests once my current PR stops churning

**Comment by SvenDowideit** [comment]
yeah, it seems to stem from the alpine blkid being much simpler than everyone else's. Time to re-write a few little things.

**Comment by SvenDowideit** [comment]
the crash installing with alpine should be fixed in v0.9.1

instead of adding netcat (for now), can you use wget?
